src/VolumeLogger.js

```diff
@@ -59,6 +59,295 @@ module.exports = class VolumeLogger {
+                            bytesReadPerSec: (metricsReport.metrics['broker/cassandra']) ? metricsReport.metrics['broker/cassandra'].readBytes : 0,
+                        },
+
+                        startTime: metricsReport.startTime,
+                        currentTime: metricsReport.currentTime,
+                        timestamp: metricsReport.currentTime,
+                    }
+
+                    if (sec === 1) {
+                        minReport.peerName = secReport.peerName
+                        minReport.startTime = secReport.startTime
+                        minReport.currentTime = secReport.currentTime
+                        minReport.timestamp = 0
+
+                        minReport.broker.messagesToNetworkPerSec = secReport.broker.messagesToNetworkPerSec
+                        minReport.broker.bytesToNetworkPerSec = secReport.broker.bytesToNetworkPerSec
+
+                        minReport.network.avgLatencyMs = secReport.network.avgLatencyMs
+                        minReport.network.bytesToPeersPerSec = secReport.network.bytesToPeersPerSec
+                        minReport.network.bytesFromPeersPerSec = secReport.network.bytesFromPeersPerSec
+                        minReport.network.connections = secReport.network.connections
+
+                        if (metricsReport.metrics['broker/cassandra']) {
+                            minReport.storage.bytesWrittenPerSec = secReport.storage.bytesWrittenPerSec
+                            minReport.storage.bytesReadPerSec = secReport.storage.bytesReadPerSec
+                        }
+                    } else {
+                        minReport.broker.messagesToNetworkPerSec = throtheledAvg(minReport.broker.messagesToNetworkPerSec, secReport.broker.messagesToNetworkPerSec)
+                        minReport.broker.bytesToNetworkPerSec = throtheledAvg(minReport.broker.bytesToNetworkPerSec, secReport.broker.bytesToNetworkPerSec)
+
+                        minReport.network.avgLatencyMs = throtheledAvg(minReport.network.avgLatencyMs, secReport.network.avgLatencyMs)
+                        minReport.network.bytesToPeersPerSec = throtheledAvg(minReport.network.bytesToPeersPerSec, secReport.network.bytesToPeersPerSec)
+                        minReport.network.bytesFromPeersPerSec = throtheledAvg(minReport.network.bytesFromPeersPerSec, secReport.network.bytesFromPeersPerSec)
+                        minReport.network.connections = throtheledAvg(minReport.network.connections, secReport.network.connections)
+
+                        if (metricsReport.metrics['broker/cassandra']) {
+                            minReport.storage.bytesWrittenPerSec = throtheledAvg(minReport.storage.bytesWrittenPerSec, secReport.storage.bytesWrittenPerSec)
+                            minReport.storage.bytesReadPerSec = throtheledAvg(minReport.storage.bytesReadPerSec, secReport.storage.bytesReadPerSec)
+                        }
+                    }
+
+                    securePublish(
```

Comment:
You need an `await` here (or `catch`) since this is an async method that could reject.

Reply:
Btw I think it would be good to move this before all the `minReport` handling above. It is a bit confusing now because the reader expects that maybe a `minReport` would get published here.